### PR TITLE
[android-auto] Search Screen

### PR DESCRIPTION
**Pull request by AndrewShkrob**
Demo:

https://user-images.githubusercontent.com/10351358/233177820-8cd59c18-51cf-4fb0-a442-1b2ecd3f1642.mp4

There are a lot of bugs on DHU's or Car Library's side. Not mine 😄 
I hope it will work correctly on real devices. I also saw the same bugs in other apps, f.e. maps.me.

Search screen:
* 0:10 - distance and opening hours are shown for the suggestion. If you check the code, you'll find it's not possible.
* 0:14 - 0:18  - distance and opening hours are displayed two times and even have different values for one search result
* 0:38 - 0:44 - search text is not updated when I click on the suggestion

SearchOnMap screen:
* same problems for distance and opening hours.

I circled all places that were displayed incorrectly with the mouse.

Also, don't check the logic of SearchOnMap screen very carefully. It's just a template and better implementation should be done. I'll prepare a separate issue for this screen.

### CODE REVIEW

File: android/src/app/organicmaps/car/screens/search/SearchOnMapScreen.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    else
+      builder.setItemList(mResults);
+    return builder.build();
+  }
+
+  @Override
+  public void onResultsUpdate(@NonNull SearchResult[] results, long timestamp)
+  {
+    if (mResults != null)
+      return;
```
Comment (biodranik) [inline android/src/app/organicmaps/car/screens/search/SearchOnMapScreen.java:71]: What's the logic here?

File: android/src/app/organicmaps/car/screens/search/SearchScreen.java [hunk 2]
```diff
@@ -35,14 +50,108 @@ public Template onGetTemplate()
...
+      invalidate();
+      return;
+    }
+
+    final MapObject location = LocationHelper.INSTANCE.getMyPosition();
+    final boolean hasLocation = location != null;
+    final double lat = hasLocation ? location.getLat() : 0;
+    final double lon = hasLocation ? location.getLon() : 0;
+
+    SearchEngine.INSTANCE.search(getCarContext(), mQuery, false, System.nanoTime(), hasLocation, lat, lon);
```
Comment (biodranik) [inline android/src/app/organicmaps/car/screens/search/SearchScreen.java:90]: Does it make sense to overload this function and simplify the calling code here and above in a similar call?

**Review by biodranik — APPROVED** [review]
LGTM, @rtsisyk PTAL

**Comment by Jean-BaptisteC** [comment]
Why sometimes two distance is present in search for one POI ?

**Comment by AndrewShkrob** [comment]
> Why sometimes two distance is present in search for one POI ?

It's a bug and not on our side 😁

File: android/src/app/organicmaps/car/screens/search/SearchOnMapScreen.java (hunk 1 above)
Comment (AndrewShkrob) [reply android/src/app/organicmaps/car/screens/search/SearchOnMapScreen.java:71]: We can't update the screen each time new results come.
We're only allowed to do it once.
We also can't cancel the search as search result points on the map will be removed.

I'll prepare a separate issue for this screen with a better description of the problem and possible solutions for it.

File: android/src/app/organicmaps/car/screens/search/SearchScreen.java (hunk 2 above)
Comment (AndrewShkrob) [reply android/src/app/organicmaps/car/screens/search/SearchScreen.java:90]: I'm not following you. What's the problem here? What can be simplified?

File: android/src/app/organicmaps/car/screens/search/SearchOnMapScreen.java (hunk 1 above)
Comment (biodranik) [reply android/src/app/organicmaps/car/screens/search/SearchOnMapScreen.java:71]: Wow, it's a big issue.

Comment (biodranik) [reply android/src/app/organicmaps/car/screens/search/SearchOnMapScreen.java:71]: We may need to change the search batching logic.

File: android/src/app/organicmaps/car/screens/search/SearchScreen.java (hunk 2 above)
Comment (biodranik) [reply android/src/app/organicmaps/car/screens/search/SearchScreen.java:90]: Create overload without haslocation and lat lon if there is no location.

**Comment by shankarpriyank** [comment]
Not related to this pull request, but in general to the development of Android Auto.
I had a look at the automotive code of PocketCast - https://github.com/Automattic/pocket-casts-android/tree/main/automotive/src/main/java/au/com/shiftyjelly/pocketcasts
They have somehow managed to implement it in a way very very similar to the normal android app development, maybe we can draw some inspiration from it

**Comment by AndrewShkrob** [comment]
>Not related to this pull request, but in general to the development of Android Auto.
I had a look at the <b>automotive</b> code of PocketCast - https://github.com/Automattic/pocket-casts-android/tree/main/automotive/src/main/java/au/com/shiftyjelly/pocketcasts
They have somehow managed to implement it in a way very very similar to the normal android app development, maybe we can draw some inspiration from it

As you said, it is <b>automotive</b> code, not Android Auto 😁 
Their code for Android Auto is located in the main module: https://github.com/Automattic/pocket-casts-android/blob/main/app/src/main/AndroidManifest.xml
As it is a media app, not a navigation one, it has a bit different interface with its quirks and features: https://developer.android.com/training/cars/media

And the link you sent is an extension of the media app for Android Automotive OS. It has nothing in common with regular android app development: https://developer.android.com/training/cars/media/automotive-os

So, no, it's impossible to use the same approach for our navigation app.

**Comment by shankarpriyank** [comment]
Yup, it is for Automotive and is a media app, maybe you are right that we can't take inspiration from their approach.
I am not updated with the latest developments in the automotive ecosystem, but the thing that fascinated me, and the main reason I shared it here was because that app is using, traditional fragments, activities, and even composable functions in the automotive code that and seemed impossible. Cause as far as I know there is a complete different way to develop UI in Android Auto and Automotive

**Review by rtsisyk — APPROVED** [review]

**Comment by rtsisyk** [comment]
> > Why sometimes two distance is present in search for one POI ?
> 
> It's a bug and not on our side 😁

Please elaborate.

**Comment by AndrewShkrob** [comment]
> > > Why sometimes two distance is present in search for one POI ?
> > 
> > 
> > It's a bug and not on our side 😁
> 
> Please elaborate.

Please take a look at the demo video and description.

**Comment by AndrewShkrob** [comment]
@biodranik @rtsisyk can we merge it?

**Pull request merged**